crates/cargo-nuget/src/main.rs

```diff
@@ -0,0 +1,274 @@
+use cargo_toml::{Manifest, Value};
+use futures::future::{BoxFuture, FutureExt};
+use structopt::StructOpt;
+use thiserror::Error;
+
+use std::ffi::OsString;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+fn main() {
+    let Opt::Nuget { subcommand } = Opt::from_args();
+    match subcommand {
+        Subcommand::Install(i) => i.perform().unwrap(),
+    }
+}
+
+/// A utility for interacting with nuget packages
+#[derive(StructOpt, Debug)]
+#[structopt(bin_name = "cargo")]
+enum Opt {
+    #[structopt(name = "nuget")]
+    Nuget {
+        #[structopt(subcommand)]
+        subcommand: Subcommand,
+    },
+}
+
+#[derive(Debug, StructOpt)]
+enum Subcommand {
+    Install(Install),
+}
+
+#[derive(Debug, Error)]
+enum Error {
+    #[error("No Cargo.toml could be found")]
+    NoCargoToml,
+    #[error("There was an error downloading the NuGet package {0}")]
+    DownloadError(Box<dyn std::error::Error>),
+    #[error("The Cargo.toml file was malformed")]
+    MalformedManifest,
+    #[error("There was some other error {0}")]
+    Other(Box<dyn std::error::Error>),
+}
+
+#[derive(Debug, StructOpt)]
+pub struct Install {}
+
+impl Install {
+    fn perform(&self) -> Result<(), Error> {
+        let bytes = std::fs::read("Cargo.toml").map_err(|_| Error::NoCargoToml)?;
+        let manifest = Manifest::from_slice(&bytes).map_err(|_| Error::MalformedManifest)?;
+        let deps = get_deps(manifest)?;
+        let downloaded_deps = download_dependencies(deps)?;
+        for dep in downloaded_deps {
+            let dep_directory = workspace_root()
+                .join("target")
+                .join("nuget")
+                .join(&dep.dependency.name);
+            // create the dependency directory
+            std::fs::create_dir_all(&dep_directory).unwrap();
+            for winmd in dep.winmds() {
+                winmd.write(&dep_directory).unwrap();
+            }
+
+            for dll in dep.dlls() {
+                dll.write(&dep_directory).unwrap();
+            }
+        }
+
+        Ok(())
+    }
+}
+
+fn workspace_root() -> PathBuf {
```

Comment:
Maybe just remove this function for now?

Reply:
I'd rather just implement it :-)